### DISCUSSION TITLE
[Question]: API Difference b/w /api/events and /api/events/summary

### DISCUSSION THREAD

**Discussion by eridon-pro** [blakeblackshear/frigate#11343] [Ask A Question] [answered]
### What is your question:

Not sure if this topic were previously covered. I would like to count detected objects during a day. Wondering what is the difference between API of events and events/summary. Here below are outputs of both. Any reason why outputs are different? Which API should I use to count the detected objects during a day? It seems sometime summary matches WebUI output but today's WebUI output doesn't match both of them.

```
$ curl -s "http://localhost:5000/api/events?cameras=CAM&after=$(date +%s -d "2024-05-12 00:00:00")&before=$(date +%s -d "2024-5-12 23:59:59")" | jq 'group_by(.label) | map({label: .[0].label, count: length})'
[
  {
    "label": "bicycle",
    "count": 21
  },
  {
    "label": "car",
    "count": 16
  },
  {
    "label": "dog",
    "count": 1
  },
  {
    "label": "motorcycle",
    "count": 1
  },
  {
    "label": "person",
    "count": 61
  }
]
```
```
$ curl -s "http://localhost:5000/api/events/summary" | jq '[.[] | select(.day == "2024-05-12" and .camera == "CAM") | {day: .day, label: .label, count: .count}]'
[
  {
    "day": "2024-05-12",
    "label": "person",
    "count": 4
  }
]
$
```

my config.yml is like this.

mqtt:
  enabled: False

go2rtc:
  streams:
    CAM: "ffmpeg:rtsp://192.168.10.209:8556/unicast#video=h264"

ffmpeg:
  hwaccel_args: preset-rpi-64-h264 #Enable Hardware Acceleration

birdseye:
  enabled: False

detectors:
  coral:
    type: edgetpu
    device: usb

record:
  enabled: True
  #enabled: False
  retain:
    days: 0
    mode: all
  events:
    pre_capture: 2
    post_capture: 2
    retain:
      default: 10
      mode: motion
      #mode: active_objects

cameras:
  CAM:
    #enabled: False
    ffmpeg:
      inputs:
      - path: rtsp://127.0.0.1:8554/CAM
        roles:
        - detect
        - record
    detect:
      enabled: True
      width: 1280 # The Cameras resolution
      height: 720 # The Cameras resolution
      fps: 10
    motion:
      mask:
      - 0,121,0,0,694,0,681,119
      - 1280,720,1280,0,963,0,551,720

objects:
  track:
  - person
  - bicycle
  - car
  - motorcycle
  - airplane
  - bus
  - car
  - skateboard
  - boat
  - bird
  - cat
  - dog
  - umbrella
  - teddy bear

**Comment by eridon-pro** [comment]
Really confused at end of the day.

$ date
Sun 12 May 23:56:32 JST 2024
$ curl -s "http://localhost:5000/api/events/summary" | jq '[.[] | select(.day == "2024-05-12" and .camera == "CAM") | {day: .day, label: .label, count: .count}]'
[
  {
    "day": "2024-05-12",
    "label": "bicycle",
    "count": 43
  },
  {
    "day": "2024-05-12",
    "label": "car",
    "count": 31
  },
  {
    "day": "2024-05-12",
    "label": "motorcycle",
    "count": 5
  },
  {
    "day": "2024-05-12",
    "label": "person",
    "count": 138
  },
  {
    "day": "2024-05-12",
    "label": "skateboard",
    "count": 1
  }
]
$ curl -s "http://localhost:5000/api/events?cameras=CAM&after=$(date +%s -d "2024-05-12 00:00:00")&before=$(date +%s -d "2024-5-12 23:59:59")" | jq 'group_by(.label) | map({label: .[0].label, count: length})'
[
  {
    "label": "bicycle",
    "count": 14
  },
  {
    "label": "car",
    "count": 18
  },
  {
    "label": "motorcycle",
    "count": 1
  },
  {
    "label": "person",
    "count": 67
  }
]
$

**Reply by NickM-27** [reply]
might be an issue with how the API is handling the timezone vs the request is sending the timezone

**Reply by eridon-pro** [reply]
I performed the API query on my Raspberry Pi 5 which hosts the Frigate. OS time zone is Asia/Tokyo. Any possibility to have different time zone at Frigate side?

**Reply by NickM-27** [reply]
> Any possibility to have different time zone at Frigate side?

That's up to you, you control if the time and what time zone is set on the container via docker

**Comment by eridon-pro** [comment]
time zone of docker of frigate and OS are same as below. Still confused which one should I use to count the objects. Does the period to search differ depending on the command (events/summary and events)?
WebUI output is almost same as "events/summary". But, not exactly same as WebUI output and API query output. It is really weird. Anything my configuration is wrong? 


$ docker ps
CONTAINER ID   IMAGE                                    COMMAND   CREATED       STATUS                  PORTS                                                                                                                NAMES
2742c1a9347c   ghcr.io/blakeblackshear/frigate:stable   "/init"   11 days ago   Up 46 hours (healthy)   0.0.0.0:5000->5000/tcp, :::5000->5000/tcp, 1935/tcp, 8555/tcp, 0.0.0.0:8554->8554/tcp, :::8554->8554/tcp, 8555/udp   frigate
$ date
Sun 19 May 21:21:58 JST 2024
$ docker container exec -it frigate date
Sun May 19 21:22:01 JST 2024
pi@EnviroPi2:~/frigate-nvr $ 


$ curl -s "http://localhost:5000/api/events/summary" | jq '[.[] | select(.day == "2024-05-19" and .camera == "CAM") | {day: .day, label: .label, count: .count}]'
[
  {
    "day": "2024-05-19",
    "label": "bicycle",
    "count": 34
  },
  {
    "day": "2024-05-19",
    "label": "car",
    "count": 39
  },
  {
    "day": "2024-05-19",
    "label": "motorcycle",
    "count": 16
  },
  {
    "day": "2024-05-19",
    "label": "person",
    "count": 121
  },
  {
    "day": "2024-05-19",
    "label": "umbrella",
    "count": 1
  }
]
$ curl -s "http://localhost:5000/api/events?cameras=CAM&after=$(date +%s -d "2024-05-19 00:00:00")&before=$(date +%s -d "2024-5-19 23:59:59")" | jq 'group_by(.label) | map({label: .[0].label, count: length})'
[
  {
    "label": "bicycle",
    "count": 9
  },
  {
    "label": "car",
    "count": 23
  },
  {
    "label": "motorcycle",
    "count": 5
  },
  {
    "label": "person",
    "count": 62
  },
  {
    "label": "umbrella",
    "count": 1
  }
]
$ curl -s "http://localhost:5000/api/events?cameras=CAM&after=$(date +%s -d "2024-05-18 15:00:00")&before=$(date +%s -d "2024-5-19 23:59:59")" | jq 'group_by(.label) | map({label: .[0].label, count: length})'
[
  {
    "label": "bicycle",
    "count": 9
  },
  {
    "label": "car",
    "count": 23
  },
  {
    "label": "motorcycle",
    "count": 5
  },
  {
    "label": "person",
    "count": 62
  },
  {
    "label": "umbrella",
    "count": 1
  }
]
$ 


$ cat docker-compose.yml 
#version: "3.9"
services:
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable
    shm_size: "128mb"
    devices:
      - /dev/bus/usb:/dev/bus/usb #Used for Coral if Available
      - /dev/video11:/dev/video11 #Map the Pi hardware acceleration
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /home/pi/frigate-nvr/config.yml:/config/config.yml
      - /home/pi/frigate-nvr/storage:/media/frigate
      - type: tmpfs #Remove this if using a Pi with 2GB or Less RAM
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "5000:5000"
      - "8554:8554" # RTSP feeds
      #- "8555:8555/tcp" # WebRTC over tcp
      #- "8555:8555/udp" # WebRTC over udp
    #environment:
    #  FRIGATE_RTSP_PASSWORD: "XXXXXX"
$ 
$ cat config.yml
mqtt:
  enabled: False

go2rtc:
  streams:
    Balcony_MJEPG_CAM: "ffmpeg:rtsp://192.168.10.209:8556/unicast#video=h264"

ffmpeg:
  hwaccel_args: preset-rpi-64-h264 #Enable Hardware Acceleration

birdseye:
  enabled: False

detectors:
  coral:
    type: edgetpu
    device: usb

record:
  enabled: True
  #enabled: False
  retain:
    days: 0
    mode: all
  events:
    pre_capture: 2
    post_capture: 2
    retain:
      default: 10
      mode: motion
      #mode: active_objects

snapshots:
  enabled: True
  timestamp: True
  bounding_box: True
  crop: True
  quality: 70
  retain:
    default: 10

timestamp_style:
  position: bl
  effect: solid

cameras:
  CAM:
    #enabled: False
    ffmpeg:
      inputs:
      - path: rtsp://127.0.0.1:8554/CAM
        roles:
        - detect
        - record
    detect:
      enabled: True
      width: 1280 # The Cameras resolution
      height: 720 # The Cameras resolution
      fps: 10
    motion:
      mask:
      - 0,121,0,0,694,0,681,119
      - 1280,720,1280,0,963,0,551,720

objects:
  track:
  - person
  - bicycle
  - car
  - motorcycle
  - airplane
  - bus
  - car
  - skateboard
  - boat
  - bird
  - cat
  - dog
  - umbrella
  - teddy bear
$

**Comment by eridon-pro** [comment, marked as answer]
Just decided to use just events API for last one hour instead of summary of the day as below. Somehow, it seems the output is accurate when it runs every 1 hour.

http://localhost:5000/api/events?cameras=$CAM&after=$(date +%s -d '1 hour ago')&before=$(date +%s)" | jq -c --arg now "$(date +'%Y/%m/%d %H:%M:%S')" '[{"Date/Time": $now}] + (group_by(.label) | map({label: .[0].label, count: length}))'
